### PR TITLE
Fix warnings caused by Ruby 2.7 update

### PR DESCRIPTION
**Pull request by taichi-ishitani**
### Summary

Fixed following warnings caused by Ruby 2.7 update.

* Using the last argument as keyword parameters is deprecated
* calling URI.open via Kernel#open is deprecated, call URI.open directly or use URI#open
* URI.escape is obsolete

### CODE REVIEW

**Comment by taichi-ishitani** [comment]
Cam anyone review my PR?

**Comment by taichi-ishitani** [comment]
Hi @chopraanmol1 , @simonoff ,
Could you please review my PR?
Thank you for your time.

**Comment by jonspalmer** [comment]
Adding my 👍  to get this reviewed and release. It a blocker for folks trying to upgrade to Ruby 2.7.x

**Comment by jonspalmer** [comment]
Maybe we should just drop Ruby 2.3 and 2.4 given they're both end of support.

**Comment by simonoff** [comment]
@jonspalmer would be great!

File: lib/roo/csv.rb [hunk 1]
```diff
@@ -90,17 +90,36 @@ def read_cells(sheet = default_sheet)
...
       ::Dir.mktmpdir(Roo::TEMP_PREFIX, ENV["ROO_TMP"]) do |tmpdir|
         tmp_filename = download_uri(filename, tmpdir)
-        ::CSV.foreach(tmp_filename, options, &block)
+        csv_foreach(tmp_filename, options, &block)
+      end
+    end
+
+    # From Ruby 2.5, options argument of CSV.new/CSV.foreach is a keyword argument.
+    # Before Ruby 2.5, that argument is a Hash.
+    # Therefore, this workaround can be removed if Ruby 2.3 and 2.4 are dropped.
```
Comment (simonoff) [inline lib/roo/csv.rb:107]: Yes, please remove such workaround.

Comment (taichi-ishitani) [reply lib/roo/csv.rb:107]: Can I edit `gemspec` and `.travis.yml` to drop ruby 2.3 & 2.4 in this branch: `ruby_27_support`?

Comment (taichi-ishitani) [reply lib/roo/csv.rb]: I removed WA for Ruby 2.3 & 2.4 and also edited `gemspec` and `.travis.yml` too.

**Comment by weilandia** [comment]
Can we get this published?? 🙏

**Pull request merged**